app/src/org/commcare/tasks/FormRecordToFileTask.java

```diff
@@ -0,0 +1,257 @@
+package org.commcare.tasks;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.os.Environment;
+import android.util.Log;
+import android.util.Pair;
+
+import org.commcare.CommCareApplication;
+import org.commcare.activities.CommCareWiFiDirectActivity;
+import org.commcare.dalvik.R;
+import org.commcare.logging.AndroidLogger;
+import org.commcare.models.database.SqlStorage;
+import org.commcare.models.database.user.models.FormRecord;
+import org.commcare.preferences.CommCarePreferences;
+import org.commcare.tasks.templates.CommCareTask;
+import org.commcare.utils.FileUtil;
+import org.commcare.utils.FormUploadUtil;
+import org.commcare.utils.SessionUnavailableException;
+import org.commcare.utils.StorageUtils;
+import org.commcare.views.notifications.NotificationMessageFactory;
+import org.commcare.views.notifications.ProcessIssues;
+import org.javarosa.core.services.Logger;
+import org.javarosa.core.services.locale.Localization;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.Properties;
+import java.util.Vector;
+
+import javax.crypto.Cipher;
+import javax.crypto.spec.SecretKeySpec;
+
+/**
+ * This class transfers all of the FormRecords from form record storage and into our file system,
+ * decrypting files as necessary. This task should really be merged with DumpTask.
+ *
+ * @author wspride
+ */
+public abstract class FormRecordToFileTask extends CommCareTask<String, String, Pair<Long, FormRecord[]>, CommCareWiFiDirectActivity> {
+    private static final String TAG = AndroidLogger.TYPE_FORM_DUMP;
+
+    private Context c;
+    private Long[] results;
+    // this is where the forms that have been pulled from FormRecord storage to the file system live
+    private File storedFormDirectory;
+
+    public static final int PULL_TASK_ID = 721356;
+
+    public FormRecordToFileTask(Context c, String formStoragePath) {
+        this.c = c;
+        this.storedFormDirectory = new File(formStoragePath);
+        taskId = PULL_TASK_ID;
+    }
+
+    @Override
+    protected void onPostExecute(Pair<Long, FormRecord[]> result) {
+        super.onPostExecute(result);
+        //These will never get Zero'd otherwise
+        c = null;
+        results = null;
+    }
+
+    private static final String[] SUPPORTED_FILE_EXTS = {".xml", ".jpg", ".3gpp", ".3gp"};
+
+    /**
+     * Turn a FormRecord folder from storage into a standard file representation in our file system.n
+     */
+    private long getFileInstanceFromStorage(File formRecordFolder, SecretKeySpec decryptionKey)
+            throws FileNotFoundException, SessionUnavailableException {

```

Comment:
`FileNotFoundException` and `SessionUnavailableException` are never thrown in this method